WordPress/gutenberg
# Update BlockPreview component to accept multiple Blocks to preview

## Description
`BlockPreview` currently only accepts a single Block. This PR updates this to allow it to (optionally) accept an array of blocks. It also simplifies the API by requiring a Block object rather than expecting Block properties as individual args. 

**Note:** the Reusable Blocks previews are not currently working due to this unrelated bug https://github.com/WordPress/gutenberg/issues/16808

## How has this been tested?
Firstly confirm existing behaviour has been preserved with the change in API. To do this, manually test all instances where `BlockPreview` is used and ensure all function as per `master` branch:
- Insert Quote block, open Block Switcher, see Previews render
- Open Block Inserter, scroll to Reusable Blocks, hover to see a preview (please see note above)

To test multiple Blocks being rendered you can:
- [Apply](https://stackoverflow.com/questions/12320863/how-do-you-take-a-git-diff-file-and-apply-it-to-a-local-branch-that-is-a-copy-o) this diff: https://gist.github.com/obenland/761894e1a5be40091e41214756b6a323
(Applying the diff is necessary to make this PR's under-the-hood changes visible)
- Create a Post with a lot of Blocks in it.
- In a quote or button block, open the Block switcher to see all of the Blocks from the page appear in the `BlockPreviewContent` below the standard Block Style previews.

![test](https://user-images.githubusercontent.com/77539/62156685-d99f6b00-b2e1-11e9-8411-58569b4956e3.gif)

### Places to check for previews

- Undo the additional diff from instructions above and test with code in this branch:
  - Block Inserter - top left of the screen the "plus" button. In the inserter, hover a reusable block (you might need to create one first) and you should see a preview on right. As noted above, you will see an error there. If you see an error, that means previews still work the same way and this PR didn't break it. *Error is good here. Seeing nothing would be bad.*
  - Block Switcher - for Button or Quote block it shows "Block Styles" as described above. Make sure both the thumbnail and the detail after hovering the thumbnail contain a preview of the style.

## Types of changes
New feature (non-breaking change which adds functionality)

## To Demo
Show multiple blocks being previewed.
